typescript/cli/src/utils/fork.ts

```diff
@@ -0,0 +1,101 @@
+import { providers } from 'ethers';
+
+import { ChainName, MultiProvider } from '@hyperlane-xyz/sdk';
+
+import { logGray, logGreen, logYellow } from '../../logger.js';
+
+import { ENV } from './env.js';
+
+const ENDPOINT_PREFIX = 'http://';
+const DEFAULT_ANVIL_ENDPOINT = 'http://127.0.0.1:8545';
+
+export enum ETH_RPC_METHODS {
+  BLOCK_NUMBER = 'eth_blockNumber',
+}
+
+export enum ANVIL_RPC_METHODS {
+  RESET = 'anvil_reset',
+  SET_RPC_URL = 'anvil_setRpcUrl',
+  SET_CHAIN_ID = 'anvil_setChainId',
+  IMPERSONATE_ACCOUNT = 'anvil_impersonateAccount',
+  NODE_INFO = 'anvil_nodeInfo',
+}
+
+/**
+ * Resets the local node to it's original start (anvil [31337] at block zero).
+ */
+export const resetFork = async () => {
+  const provider = getLocalProvider();
+  await provider.send(ANVIL_RPC_METHODS.RESET, []);
+};
+
+/**
+ * Forks a chain onto the local node at the latest block of the forked network.
+ * @param multiProvider the multiProvider with which to fork the network
+ * @param chain the network to fork
+ */
+export const setFork = async (
+  multiProvider: MultiProvider,
+  chain: ChainName | number,
+) => {
+  logGray(`Forking ${chain} for dry-run...`);
+
+  const provider = getLocalProvider();
+  const currentChainMetadata = multiProvider.metadata[chain];
+
+  await provider.send(ANVIL_RPC_METHODS.SET_CHAIN_ID, [
+    currentChainMetadata.chainId,
+  ]);
```

Comment:
do we even need to do this? I wasn't doing this in the other fork scripts iirc

Reply:
No we do not, and this is also post-wip cleanup. Now removed